lib/bulk_generate.rb

```diff
@@ -16,26 +21,31 @@ def trns
   end
 
   def ects
-    @ects ||= trns.shuffle.take(sample_size)
-  end
-
-  def columns_for(template)
-    template.values.reject { |v| v.match?(/error/i) }
+    @ects ||= trns.map do |trn, date_of_birth|
+      status = api_client.find_teacher(trn:, date_of_birth:).present[:trs_induction_status]
+      [trn, date_of_birth, status]
+    rescue StandardError
+      [trn, date_of_birth, 'API could not be contacted']
+    end
   end
 
   def claim_rows
-    ects.map do |trn, dob, _status|
-      [trn, dob, random_programme, random_date]
+    ects.map do |trn, date_of_birth, status|
+      [trn, date_of_birth, random_programme, random_start_date, status]
     end
   end
 
   def action_rows
-    ects.map do |trn, dob, _status|
-      [trn, dob, random_date, random_term, random_outcome]
+    ects.map do |trn, date_of_birth, status|
+      [trn, date_of_birth, random_end_date, random_term, random_outcome, status]
     end
   end
 
-  def random_date
+  def random_start_date
+    61.days.ago.to_date - rand(1..60)
```

Comment:
```suggestion
    (rand(1..60)).days.ago.to_date
```

Reply:
@peteryates I actioned this if you have time to look at the newest commit